org.eclipse.eclemma.ui.test/src/org/eclipse/eclemma/ui/DumpExecutionDataTest.java

```diff
@@ -0,0 +1,144 @@
+/*******************************************************************************
+ * Copyright (c) 2006, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ ******************************************************************************/
+package org.eclipse.eclemma.ui;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.core.runtime.ILogListener;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.eclemma.core.CoverageTools;
+import org.eclipse.eclemma.core.ISessionManager;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable;
+import org.eclipse.swtbot.swt.finder.results.VoidResult;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class DumpExecutionDataTest {
+
+	private static final SWTWorkbenchBot bot = new SWTWorkbenchBot();
+
+	private static ILaunch launch1;
+	private static ILaunch launch2;
+
+	@AfterClass
+	public static void resetWorkbench() {
+		bot.resetWorkbench();
+	}
+
+	@BeforeClass
+	public static void setup() throws Exception {
+		openCoverageView();
+
+		JavaProjectKit project = new JavaProjectKit();
+		project.enableJava5();
+		final IPackageFragmentRoot root = project.createSourceFolder();
+		final IPackageFragment fragment = project.createPackage(root, "example");
+		project.createCompilationUnit(fragment, "Example.java",
+				"package example; class Example { public static void main(String[] args) throws Exception { Thread.sleep(30000); } }");
+
+		JavaProjectKit.waitForBuild();
+
+		ILaunchConfiguration launchConfiguration = createLaunchConfiguration(project.javaProject, "example.Example");
+
+		launch1 = launchConfiguration.launch(CoverageTools.LAUNCH_MODE, null);
+		launch2 = launchConfiguration.launch(CoverageTools.LAUNCH_MODE, null);
+	}
+
+	@Test
+	public void test() throws Exception {
```

Comment:
It might be hard to tell in future why this test case has been created. We could improve documentation here with the more descriptive test case names like we use in JaCoCo, e.g.

    dumpDialog_should_abort_without_errors_when_cancel_is_pressed

Also I would prefer to move the positive case into a separate test case.